src/Common.am

```diff
@@ -73,3 +76,20 @@ COMPAT_LIB = $(top_builddir)/compat/libcompatsquid.la $(LIBPROFILER)
 
 ## Some helpers are written in Perl and need the local shell defined properly
 subst_perlshell = sed -e 's,[@]PERL[@],$(PERL),g' <$(srcdir)/$@.pl.in >$@ || ($(RM) -f $@ ; exit 1)
+
+.h.hdrtest:
+	@SrcFilePath=`echo $< | $(SED) 's%^$(top_srcdir)/%%'`; \
+	SrcFileName=`basename $$SrcFilePath`; \
+	TargetFileName=`basename $@`; \
+	if test "$(EXCLUDE_FROM_HDR_TESTING)" != "" && echo "$(EXCLUDE_FROM_HDR_TESTING)" | $(TR) ' ' '\n' | $(FGREP) -x -q "$$SrcFileName" ; then \
+		echo "header-test: skip - $$SrcFilePath"; \
+		exit 0; \
+	fi; \
+	$(SED) "s%[@]HEADER[@]%${<}%" "$(top_srcdir)/test-suite/testHeader.cc.in" >"$$TargetFileName.cc" && \
+	echo "$(CXXCOMPILE) -c -o" "$$TargetFileName" "$$TargetFileName.cc" && \
+	if $(CXXCOMPILE) -c -o "$$TargetFileName" "$$TargetFileName.cc" ; \
+	then echo "header-test: ok - $$SrcFilePath"; $(RM) "$$TargetFileName.cc" "$$TargetFileName"; \
+	else echo "header-test: not ok - $$SrcFilePath"; exit 1; fi
+
+# nothing to do, skip
+.cc.hdrtest:
```

Comment:
Can you explain why we need this `.cc.hdrtest` target (but not, say, `.cci.hdrtest` target), please?

----

```
testHeaders: ...
    $(MAKE) $(^:.h=.hdrtest) && cp $(TRUE) $@
```

AFAICT, we need a `.h.hdrtest` target because testHeaders targets are calling `make` with foo.hdrtest names and, without that target, `make foo.hdrtest` would not know how to make foo.hdrtest. That reasoning does not explain the need for `.cc.hdrtest` target though.

```
libhttp_la_SOURCES = \
    ...
    RegisteredHeadersHash.cci \
```

And if `.cc.hdrtest` target must exist, then I still do not understand why there are no targets for other extensions in testHeaders dependencies (e.g., `.cci` and `.in`).

Reply:
It's actually not needed; .cc files will escape the $(^:.h=.hdrtest) conversion and thus appear as already made by make. Removing.